### ISSUE TITLE
Build fails on nixos-unstable

### ISSUE THREAD

**Issue by SuperSandro2000** [Gabriella439/nix-diff#72]
```
nix-diff> Preprocessing library for nix-diff-1.0.18..
nix-diff> Building library for nix-diff-1.0.18..
nix-diff> [1 of 4] Compiling Nix.Diff.Types   ( src/Nix/Diff/Types.hs, dist/build/Nix/Diff/Types.o, dist/build/Nix/Diff/Types.dyn_o )
nix-diff>
nix-diff> src/Nix/Diff/Types.hs:103:12: error:
nix-diff>     • solveWanteds: too many iterations (limit = 4)
nix-diff>         Unsolved: WC {}
nix-diff>         Set limit with -fconstraint-solver-iterations=n; n=0 for no limit
nix-diff>     • When deriving the instance for (Arbitrary DerivationDiff)
nix-diff>     |
nix-diff> 103 |   deriving Arbitrary via GenericArbitrary DerivationDiff
nix-diff>     |            ^^^^^^^^^
```

I also needed to apply the following diff:

```diff
diff --git a/flake.lock b/flake.lock
index a1ae9f9..09715cf 100644
--- a/flake.lock
+++ b/flake.lock
@@ -2,16 +2,16 @@
   "nodes": {
     "nixpkgs": {
       "locked": {
-        "lastModified": 1672441588,
-        "narHash": "sha256-jx5kxOyeObnVD44HRebKYL3cjWrcKhhcDmEYm0/naDY=",
+        "lastModified": 1672525397,
+        "narHash": "sha256-WASDnyxHKWVrEe0dIzkpH+jzKlCKAk0husv0f/9pyxg=",
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "6a0d2701705c3cf6f42c15aa92b7885f1f8a477f",
+        "rev": "8ba56d7c0d7490680f2d51ba46a141eca7c46afa",
         "type": "github"
       },
       "original": {
         "owner": "NixOS",
-        "ref": "nixos-22.11",
+        "ref": "nixos-unstable",
         "repo": "nixpkgs",
         "type": "github"
       }
diff --git a/flake.nix b/flake.nix
index 8c60dca..ad27c6b 100644
--- a/flake.nix
+++ b/flake.nix
@@ -1,5 +1,5 @@
 { inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";

     utils.url = "github:numtide/flake-utils";
   };
diff --git a/nix-diff.cabal b/nix-diff.cabal
index 1d32735..cef8d71 100644
--- a/nix-diff.cabal
+++ b/nix-diff.cabal
@@ -52,7 +52,7 @@ executable nix-diff
                      , nix-diff
                      , aeson
                      , bytestring
-                     , optparse-applicative >= 0.14.0.0 && < 0.17
+                     , optparse-applicative >= 0.14.0.0 && < 0.18
                      , text
                      , unix                                < 2.8
                      , containers
```

**Comment by Gabriella439** [comment]
It seems to be due to this: https://gitlab.haskell.org/ghc/ghc/-/issues/22516

**Comment by maralorn** [comment]
> It seems to be due to this: https://gitlab.haskell.org/ghc/ghc/-/issues/22516

That issue claims that this is regression from 9.2 to 9.4, but we have this build problem with 9.2. Not saying you are wrong, just that something doesn‘t seem to add up.

**Comment by Player-205** [comment]
Fixed via #74
